src/components/DataProvider/index.js

```diff
@@ -131,6 +131,14 @@ export default class DataProvider extends Component {
         this.startUpdateInterval();
       }
     }
+
+    // check if pointsPerSeries changed in props -- if so fetch new data
+    if (this.props.pointsPerSeries !== prevProps.pointsPerSeries) {
+      await Promise.map(this.props.series, s =>
+        this.fetchData(s.id, 'UPDATE_POINTS_PER_SERIES')
+      );
```

Comment:
Should there be an early-return after this? I'm not sure what the consequences are of continuing with this method.

@valerii-cognite @martincognite

Reply:
This needs to be better documented (my bad i guess), but after going through it we already do fetch data twice, if there are new series and the `baseDomain` have changed in the same update cycle.

I would really like to see a story on this that logs the actions to make sure we all understand what's going on (re-use the `LineChart/basic` story and dynamically change `pointsPerSeries`).